src/DssProxyActionsCropper.sol

```diff
@@ -629,43 +626,41 @@ contract DssProxyActionsCharter is Common {
         address owner = CdpRegistryLike(cdpRegistry).owns(cdp);
         bytes32 ilk = CdpRegistryLike(cdpRegistry).ilks(cdp);
 
-        address urp = CharterLike(charter).getOrCreateProxy(owner);
+        address urp = CropperLike(cropper).getOrCreateProxy(owner);
         (, uint256 art) = VatLike(vat).urns(ilk, urp);
 
         // Joins DAI amount into the vat
         daiJoin_join(daiJoin, owner, _getWipeAllWad(owner, urp, ilk));
-        // Allows charter to access to proxy's DAI balance in the vat
-        VatLike(vat).hope(charter);
+        // Allows cropper to access to proxy's DAI balance in the vat
+        VatLike(vat).hope(cropper);
         // Paybacks debt to the CDP and unlocks token amount from it
         _frob(ilk, owner, -_toInt256(_convertTo18(gemJoin, amtC)), -_toInt256(art));
-        // Denies charter to access to proxy's DAI balance in the vat after execution
-        VatLike(vat).nope(charter);
+        // Denies cropper to access to proxy's DAI balance in the vat after execution
+        VatLike(vat).nope(cropper);
+        // Exits token amount to proxy address as a token
+        CropperLike(cropper).exit(gemJoin, address(this), amtC);
         // Exits token amount to the user's wallet as a token
-        CharterLike(charter).exit(gemJoin, msg.sender, amtC);
+        GemJoinLike(gemJoin).gem().transfer(msg.sender, amtC);
     }
 
-    function roll(
-        uint256 srcCdp,
-        uint256 dstCdp,
-        uint256 wad
+    function crop(
+        address gemJoin,
+        uint256 cdp
     ) external {
-        bytes32 srcIlk = CdpRegistryLike(cdpRegistry).ilks(srcCdp);
+        address owner = CdpRegistryLike(cdpRegistry).owns(cdp);
+        require(CdpRegistryLike(cdpRegistry).ilks(cdp) == GemJoinLike(gemJoin).ilk(), "wrong-ilk");
 
-        // Gets actual rate from the vat
-        (, uint256 rate,,,) = VatLike(vat).ilks(srcIlk);
-        CharterLike(charter).roll(
-            srcIlk,
-            CdpRegistryLike(cdpRegistry).ilks(dstCdp),
-            CdpRegistryLike(cdpRegistry).owns(srcCdp),
-            CdpRegistryLike(cdpRegistry).owns(dstCdp),
-            _mul(wad, RAY) / rate
-        );
+        CropperLike(cropper).join(gemJoin, owner, 0);
+        GemLike bonus = GemJoinLike(gemJoin).bonus();
+        bonus.transfer(msg.sender, bonus.balanceOf(address(this)));
     }
+
+    // TODO: do we want to support also flee()?
```

Comment:
I don't see any harm in adding this support even though it's not really something that should be used unless there is an emergency.

Reply:
@hexonaut do you if there is a specific reason why in cropper's flee() there is no `usr` parameter? I think it should get one like we have in exit().
This will now also save gas in the new action fleeGem(), as we can flee directly to the original msg.sender.

Reply:
anyway I added it.